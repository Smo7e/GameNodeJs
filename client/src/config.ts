// UdSU
// const DOMAIN = 'http://studfront';
// const PORT = 81;

// dev
const DOMAIN = "http://localhost";
const PORT = 3001;

// prod
//...

export const HOST = PORT ? `${DOMAIN}:${PORT}` : `${DOMAIN}`;

export const MEDIATOR = {
    EVENTS: {
        SERVER_ERROR: "SERVER_ERROR",
        GET_MESSAGES: "GET_MESSAGES",
        GET_SCENE: "GET_SCENE",
        GET_USER: "GET_USER",
        GET_GAMERS: "GET_GAMERS",
        GET_FRIENDS: "GET_FRIENDS",
        GET_INVITES: "GET_INVITES",
        LOGIN: "LOGIN",
        SIGNUP: "SIGNUP",
        LOGOUT: "LOGOUT",
        GET_MOBS: "GET_MOBS",
        GET_ERROR: "GET_ERROR",
        PLAY_MUSIC: "PLAY_MUSIC", 
        STOP_MUSIC: "STOP_MUSIC", 
        SET_MUSIC_VOLUME: "SET_MUSIC_VOLUME", 
        UPDATE_ARR_BULLET_TRAJECTORY: "UPDATE_ARR_BULLET_TRAJECTORY",
    },
    TRIGGERS: {},
};
