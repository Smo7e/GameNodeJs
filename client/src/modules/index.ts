import { TError } from "./Server/types";
import Server from "./Server/Server";
import Mediator from "./Mediator/Mediator";
import Store from "./Store/Store";

export type { TError };
export { Server, Mediator, Store };
