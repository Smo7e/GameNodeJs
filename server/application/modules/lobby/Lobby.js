class Lobby {
    constructor(answer, db, io, mediator) {
        this.answer = answer;
        this.db = db;
        this.io = io;
        this.mediator = mediator;
        if (!io) {
            return;
        }
        this.invites = {};

        io.on("connection", (socket) => {
            socket.on("GET_ITEMS", () => this.getItems(socket.id));
            socket.on("ADD_INVITES", (data) => this.addInvitation(data, socket.id));
            socket.on("GET_INVITES", (data) => this.checkInvites(data, socket.id));
        });
    }
    async getItems() {
        const item = await this.db.getItems();
        if (item) {
            this.io.emit("GET_ITEMS", this.answer.good(item));
            return;
        }
        this.io.emit("GET_ITEMS", this.answer.bad(405));
        return;
    }

    async addInvitation({ userId, friendId, lobbyName }) {
        if (!this.invites[friendId]) {
            this.invites[friendId] = {
                friendsId: [],
                lobbyName: null,
            };
        }
        this.invites[friendId].friendsId.push(userId);
        this.invites[friendId].lobbyName = lobbyName;
        this.io.emit("GET_INVITES", this.answer.good(this.invites[friendId]));
    }
    async checkInvites({ userId }) {
        const invites = this.invites[userId];
        if (invites) {
            this.io.emit("GET_INVITES", this.answer.good(invites));
            return;
        }
    }
}
module.exports = Lobby;
