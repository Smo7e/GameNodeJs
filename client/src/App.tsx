import React, { useState } from "react";
import { HOST, MEDIATOR } from "./config";
import { Server, Mediator, Store, TError } from "./modules";
import SignUp from "./component/SignUp/SignUp";
import Game from "./modules/Game/Game";
import Menu from "./component/Menu/Menu";
import Login from "./component/Login/Login";
import Heroes from "./component/Heroes/Heroes";
import Interface from "./component/Interface/Interface";
import Lobby from "./component/Lobby/Lobby";
import ErrorMessage from "./modules/ErrorMessage/ErrorMessage";
import Parameters from "./component/Parameters/Parameters";
import Music from "./component/Parameters/Music";
import "./App.css";

export const ServerContext = React.createContext<Server>(null!);
export const MediatorContext = React.createContext<Mediator>(null!);
export const StoreContext = React.createContext<Store>(null!);

export enum EPAGES {
    SIGNUP,
    LOGIN,
    GAME,
    MENU,
    HEROES,
    LOBBY,
    PARAMETERS,
}

const MainApp = () => {
    const [epages, setEpages] = useState<EPAGES>(EPAGES.LOGIN);
    return (
        <>
            {epages === EPAGES.LOGIN ? (
                <Login epages={setEpages} />
            ) : epages === EPAGES.SIGNUP ? (
                <SignUp epages={setEpages} />
            ) : epages === EPAGES.MENU ? (
                <Menu epages={setEpages} />
            ) : epages === EPAGES.GAME ? (
                <>
                    <Game />
                    <Interface />
                </>
            ) : epages === EPAGES.HEROES ? (
                <Heroes epages={setEpages} />
            ) : epages === EPAGES.LOBBY ? (
                <Lobby epages={setEpages} />
            ) : epages === EPAGES.PARAMETERS ? (
                <Parameters epages={setEpages} />
            ) : (
                <></>
            )}
            <ErrorMessage />
            <Music />
        </>
    );
};

const App: React.FC = () => {
    const store = new Store();
    const mediator = new Mediator(MEDIATOR);
    const server = new Server(HOST, mediator, store);

    const { SERVER_ERROR } = mediator.getEventTypes();
    mediator.subscribe(SERVER_ERROR, (data: TError) => {
        console.log(data);
    });

    return (
        <StoreContext.Provider value={store}>
            <MediatorContext.Provider value={mediator}>
                <ServerContext.Provider value={server}>
                    <MainApp />
                </ServerContext.Provider>
            </MediatorContext.Provider>
        </StoreContext.Provider>
    );
};

export default App;
