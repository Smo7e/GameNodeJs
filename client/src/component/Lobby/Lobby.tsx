import React, { useState, useEffect, useRef, useContext } from "react";
import { EPAGES, MediatorContext, ServerContext, StoreContext } from "../../App";
import SportikLobby from "./component/SportikLobby";
import HumanitarianLobby from "./component/HumanitarianLobby";
import TechguyLobby from "./component/TechguyLobby";
import ShopLobby from "./component/Shop";

import Addafriend1 from "./panel/Addafriend1";
import Addafriend2 from "./panel/Addafriend2";

import "./Lobby.css";
import FriendLobby1 from "./component/FriendLobby1";
import FriendLobby2 from "./component/FriendLobby2";
import { VARIABLE } from "../../modules/Store/Store";

interface ILobbyProps {
    epages: Function;
}

export enum ELOBBY {
    SPORTIK,
    HUMANITARIAN,
    TECHGUY,
}

export enum EPANEL {
    ADDAFRIEND1,
    ADDAFRIEND2,
}

const Lobby: React.FC<ILobbyProps> = ({ epages }) => {
    const mediator = useContext(MediatorContext);
    const store = useContext(StoreContext);

    const [lobby, setLobby] = useState<ELOBBY>(ELOBBY.SPORTIK);
    const [panel, setPanel] = useState<EPANEL>();
    const [gamers, setGamers] = useState<any>([]);
    const panelRef = useRef<HTMLDivElement>(null);

    const gameHadler = async () => {
        epages(EPAGES.GAME);
    };

    useEffect(() => {
        const handleClickOutside = (event: MouseEvent) => {
            if (panelRef.current && !panelRef.current.contains(event.target as Node)) {
                setPanel(undefined);
            }
        };
        const { GET_GAMERS } = mediator.getEventTypes();

        const getGamersHandler = (data: any) => {
            setGamers(data);
        };
        mediator.subscribe(GET_GAMERS, getGamersHandler);
        document.addEventListener("mousedown", handleClickOutside);
        return () => {
            document.removeEventListener("mousedown", handleClickOutside);
            mediator.unsubscribe(GET_GAMERS, getGamersHandler);
        };
    }, []);
    if (!gamers || !gamers[0] || !gamers[0].name) return <></>;
    ///ПЕРЕДЕЛАТЬ НЕРИАЛЬНЫХ КРИНЖ
    return (
        <div id="test-container-Lobby" className="container-Lobby">
            <button onClick={() => epages(EPAGES.MENU)} id="test-arrow-1" className="arrow-1"></button>
            <ShopLobby />
            <FriendLobby1 gamers={gamers} setPanel={setPanel} />
            <FriendLobby2 gamers={gamers} setPanel={setPanel} />
            {lobby === ELOBBY.SPORTIK && store.get(VARIABLE.GAMER).post === "Admin" ? (
                <SportikLobby lobby={setLobby} gamerNumber={0} />
            ) : lobby === ELOBBY.HUMANITARIAN && store.get(VARIABLE.GAMER).post === "Admin" ? (
                <HumanitarianLobby lobby={setLobby} gamerNumber={0} />
            ) : lobby === ELOBBY.TECHGUY && store.get(VARIABLE.GAMER).post === "Admin" ? (
                <TechguyLobby lobby={setLobby} gamerNumber={0} />
            ) : (
                <>
                    {gamers && gamers[0] && `${gamers[0].person_id}` && gamers[0].person_id - 0 === 0 ? (
                        <div className="image-Sportik">
                            <button className="button">&lt;Спортик&gt;</button>
                        </div>
                    ) : gamers && gamers[0] && `${gamers[0].person_id}` && gamers[0].person_id - 0 === 1 ? (
                        <div className="image-techguy">
                            <button className="button">&lt;Технарь&gt;</button>
                        </div>
                    ) : gamers && gamers[0] && `${gamers[0].person_id}` && gamers[0].person_id - 0 === 2 ? (
                        <div className="image-humanitarian">
                            <button className="button">&lt;Гуманитарий&gt;</button>
                        </div>
                    ) : (
                        <>чо то не то</>
                    )}
                </>
            )}
            {panel === EPANEL.ADDAFRIEND1 ? (
                <div ref={panelRef} id="test-Addafriend1">
                    <Addafriend1 friends={store.get(VARIABLE.FRIENDS)} userId={store.get(VARIABLE.USER).id} />
                </div>
            ) : panel === EPANEL.ADDAFRIEND2 ? (
                <div ref={panelRef} id="test-Addafriend2">
                    <Addafriend2 friends={store.get(VARIABLE.USER).id} userId={store.get(VARIABLE.USER).id} />
                </div>
            ) : (
                <></>
            )}
            <button onClick={gameHadler} id="test-play" className="play">
                ИГРАТЬ
            </button>
        </div>
    );
};

export default Lobby;
