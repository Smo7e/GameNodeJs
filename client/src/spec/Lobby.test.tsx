import { render, screen } from "@testing-library/react";
import Lobby from "../component/Lobby/Lobby";
import { MediatorContext, ServerContext } from "../App";
import { Mediator, Server, Store } from "../modules";
import { HOST, MEDIATOR } from "../config";

const store = new Store();

const mediator = new Mediator(MEDIATOR);
const server = new Server(HOST, mediator, store);

test("Lobby snapshot", () => {
    const { asFragment } = render(
        <MediatorContext.Provider value={mediator}>
            <ServerContext.Provider value={server}>
                <Lobby epages={() => {}} />
            </ServerContext.Provider>
        </MediatorContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});
