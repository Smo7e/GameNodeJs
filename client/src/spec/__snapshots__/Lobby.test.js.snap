// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lobby snapshot 1`] = `<DocumentFragment />`;
